### ISSUE TITLE
Proposal: Provide simple benchmark example

### ISSUE THREAD

**Issue by thought2** [mtkennerly/kelonio#9]
I think it could make sense to provide a very simple example that shows how to *compare* the performance of two syncronous functions. Maybe taking the same or a similar one (like "benchmark.js")[https://github.com/bestiejs/benchmark.js#installation].

In kelonio, the first example provided shows how to measure one function. The following ones are testing library specific.

What do you think?

**Comment by mtkennerly** [comment]
I'm open to adding that. Do you mean an example like this one from the Benchmark.js docs?

```javascript
.on('complete', function() {
  console.log('Fastest is ' + this.filter('fastest').map('name'));
})
```

Just working with `Measurement`s directly, you could do something like:

```typescript
const measurements: { [key: string]: Measurement } = {
    "RegExp#test": await measure(() => /o/.test("Hello World")),
    "String#indexOf": await measure(() => "Hello World!".indexOf("o") > -1),
};
console.log(`Fastest: ${Object.entries(measurements).reduce((a, b) => a[1].mean < b[1].mean ? a : b)[0]}`);
```

You can do that with a full set of results by inspecting `benchmark.data`, but It's a bit more cumbersome since you have to recurse through the nested levels. It might be worth adding a wrapper around that to loop through all of the results without worrying about the nesting or having to convert the raw duration arrays into `Measurement`s, so that you can focus on just filtering by your criteria. I'll think over some potential enhancements here before updating the docs.

**Comment by MatheusRich** [comment]
+1 for this.

**Comment by mtkennerly** [comment]
Just an update: I've added some helper methods to make this easier, but the reason it's taking a while to release is because I still need to figure out how best to let users invoke these helpers as part of the final report when operating inside of a test framework. My first idea was to allow configuring a reporter callback, which would work for Jest and Mocha, but I think the Karma config only allows plain data. I'm currently considering a configurable Nunjucks template, but I still need to try it out and see how well it works.

In the meantime, this is how it looks when used on its own:

```typescript
import { Benchmark, Criteria } from "kelonio";

const benchmark = new Benchmark();
await benchmark.record("RegExp#test", () => /o/.test("Hello World"));
await benchmark.record("String#indexOf", () => "Hello World!".indexOf("o") > -1);

const fastest = benchmark.find(Criteria.Fastest);
console.log(`Fastest: ${fastest?.description} with mean ${fastest?.mean} ms`);
// Fastest: String#indexOf with mean 0.004199049999999999 ms
```

**Comment by mtkennerly** [comment]
Sorry for the wait, but this is finally available now in v0.8.0! Each reporter has an `extensions` option, allowing you to configure custom report functions.

For example, in jest.config.js, add the reporter option:

```javascript
[
    "<rootDir>/node_modules/kelonio/out/plugin/jestReporter",
    {
        extensions: [
            { module: `${__dirname}/extension.js`, extension: "extension" },
        ],
    },
]
```

And then in `extension.js`:

```javascript
const { Criteria } = require("kelonio");

module.exports = {
    extension: {
        extraReport: benchmark => {
            const fastest = benchmark.find(Criteria.Fastest);
            if (fastest) {
                return `
= = Custom Report = =
Fastest: "${fastest.description.join("/")}" (${fastest.mean} ms)
= = = = = = = = = = =
        `;
            }
        }
    }
};
```
